src/main/java/org/apache/flink/benchmark/SerializationFrameworkMiniBenchmarks.java

```diff
@@ -128,6 +146,50 @@ public void serializerRow() throws Exception {
 		env.execute();
 	}
 
+	/**
+	 * Source emitting a simple {@link MyPojo POJO}, but with long random strings.
+	 */
+	public static class StringHeavyPojoSource extends BaseSourceWithKeyRange<MyPojo> {
+		private static final long serialVersionUID = 2941333602938145526L;
+
+		private transient MyPojo template;
+
+		public StringHeavyPojoSource(int numEvents, int numKeys) {
+			super(numEvents, numKeys);
+		}
+
+		@Override
+		protected void init() {
+			super.init();
+			template = new MyPojo(
+					0,
+					makeString(StringSerializationBenchmark.asciiChars, 128),
```

Comment:
If we bump the string length to 1024, does that shows bigger difference for between and after the improvement? or even higher, like 4KB? 

It would be better to have this single benchmark. Adding more benchmarks (like `StringSerializationBenchmark` or `PojoSerializationBenchmark`) has to be carefully weighted as the trade off is longer benchmarking times.

Some partial solution might be to move those two `StringSerializationBenchmark` and `PojoSerializationBenchmark` benchmarks to `full` package.

Reply:
If we want to focus on the core of the contribution, `StringSerializationBenchmark` is the best way to see the impact. However, the continuously running benchmark suite is mostly for finding regressions and I doubt that we often change the string serialization, such that this benchmark will stay constant most of the time. Thus, I like the idea of moving that test to to `full`, so we can use it to drill down investigations if the serialization indeed worsens.

The same is true for `PojoSerializationBenchmark`, although it has a broader scope and could also show regressions in other serialization code paths. Nevertheless, both tests are superseded by this `SerializationFrameworkMiniBenchmarks`, which is enough to indicate issues in serialization and thus should be the only benchmark not in `full`. 

It would be awesome to just have one test, where we see a big impact by FLINK-14346. Imho the current 20% is a huge improvement, but as @pnowojski pointed out, it may be even more obvious if you could increase the size for this one benchmark to even some higher values.

Reply:
@shuttie re-opening the issue.
> see comments for the original PR for huge string benchmarks. TLDR: they are as good as for smaller strings.

I wasn't concerned with improvement not being visible on huge strings. On the contrary. I was hoping that with larger strings, the performance improvements that you made will be even more visible (for example +50% instead of +20%), out-weighting other overheads of `SerializationFrameworkMiniBenchmarks#serializerStringHeavyPojo`, so testing even better for the string serialiser changes.

@AHeise agree, we can have more detailed benchmarks in `full` package even if tweak this benchmark to stress more string serialiser.

Reply:
Okay then one last round to build on what @pnowojski . The full benchmarks are good as they are. The question is can we make the improvements in this one test `serializerStringHeavyPojo` more pronounced.
I'm currently thinking of simplifying the test by simply using a `LongStringSource` that only returns strings of size 1024. Then we would get rid of the constant tuple serialization overhead.

A rough sketch of the source:
```java
	/**
	 * Source emitting a long String.
	 */
	public static class LongStringSource extends BaseSourceWithKeyRange<String> {
		private static final long serialVersionUID = 3746240885982877398L;
		private String[] templates;

		public LongStringSource(int numEvents, int numKeys) {
			super(numEvents, numKeys);
		}

		@Override
		protected void init() {
			super.init();
			templates = new String[] {
				makeString(StringSerializationBenchmark.asciiChars, 1024),
				makeString(StringSerializationBenchmark.russianChars, 1024),
				makeString(StringSerializationBenchmark.chineseChars, 1024)
			};
		}

		private String makeString(char[] symbols, int length) {
			char[] buffer = new char[length];
			Random random = ThreadLocalRandom.current();
			Arrays.fill(buffer, symbols[random.nextInt(symbols.length)]);
			return new String(buffer);
		}

		@Override
		protected String getElement(int keyId) {
			return templates[keyId % templates.length];
		}
	}
```

Reply:
@pnowojski the main problem with the original benchmark code was using parallelism=4. It measured memory bandwidth instead of the actual code performance. See the last comment in this PR for the details.